tests/test__collection_api.py

```diff
@@ -4682,6 +4677,32 @@ def test__set_union(self):
         }]
         self.assertEqual(expect, list(actual))
 
+    def test__set_equals(self):
+        collection = self.db.collection
+        collection.insert_many([
+            {'array': ['one', 'three']},
+        ])
+        actual = collection.aggregate([{'$project': {
```

Comment:
Please also add an equivalent case in test__mongomock to make sure the behavior is the same than the one of pymongo.

Reply:
Done!